### PR TITLE
dev/ci: explicitly fetch dogfood cluster's credentials in this repo's CI

### PR DESCRIPTION
**Pull request by ggilmore**
Follow up to: https://github.com/sourcegraph/infrastructure/pull/659

### CODE REVIEW

File: dev/ci/gen-pipeline.go [hunk 1]
```diff
@@ -213,6 +217,7 @@ func main() {
 			bk.Env("VERSION", version),
 			bk.Env("CONTEXT", "gke_sourcegraph-dev_us-central1-a_dogfood-cluster-7"),
 			bk.Env("NAMESPACE", "default"),
```
Comment (slimsag) [inline dev/ci/gen-pipeline.go]: Do we still need these env vars? If you're not sure, a TODO is sufficient.

Comment (ggilmore) [reply dev/ci/gen-pipeline.go]: I am not sure about "VERSION", but I like being explicit with `CONTEXT` and `NAMESPACE` for the reason that I mentioned the linked issue

Comment (ggilmore) [reply dev/ci/gen-pipeline.go]: I like that https://github.com/sourcegraph/sourcegraph/blob/master/dev/ci/deploy-dogfood.sh is written in such a way that all the `kubectl` commands are nicely scoped

Comment (slimsag) [reply dev/ci/gen-pipeline.go]: 👍

**Pull request merged**